### ISSUE TITLE
Space keypress on a focused <button /> is swallowed by useKeyPress

### ISSUE THREAD

**Issue by seeshinghei** [xyflow/xyflow#5037]
### What platform were you using when you found the bug?

- React Flow version: 12.0.4
- Browser and version: Google Chrome 131.0.6778.265
- OS and version: MacOS Sequoia 15.2

### Live code example

https://codesandbox.io/p/sandbox/happy-gauss-cldqm6?file=%2Fsrc%2FApp.tsx%3A28%2C7

### Describe the Bug

Space key to trigger onClick action of a focused `<button />` is swallowed by useKeyPress when `<ReactFlow />` is used.

When the space key is pressed, the event.preventDefault() stops the default behaviour of triggering the onClick action of a focused button.
https://github.com/xyflow/xyflow/blob/e041720a419c7248efc3603d24b4b5e12cf85978/packages/react/src/hooks/useKeyPress.ts#L102-L105

### Steps to reproduce the bug or issue

1. Create a `<button />` with an onClick action
2. Focus on the button with keyboard tab
3. Press space

### Expected behavior

As a user, I should be able to trigger the action of an ordinary button by focusing the button and press tab, which does the same thing as pressing enter, but the space keypress is not responding.

### Additional context

If the space is pressed very quickly right after the tab to focus (or pressed together with the tab), the onClick action can be triggered.

**Comment by bcakmakoglu** [comment]
As a workaround you can add `className="nokey"` to elements that should be ignored.

**Comment by seeshinghei** [comment]
Thanks @bcakmakoglu for the quick help. The suggestion of adding a className `nokey` makes the button responding to a space keypress.

Wondering if it is a temporary workaround and subject to change in the future as I look into the implementation of `nokey` in `isInputDOMNode` that is is more for checking the DOM Node being a input or not. The official site also did not mention the usage of `nokey`. With `nokey`, we are skipping the elements that we do not want to go through the `useKeyPress` logic.

**Comment by lukebennett88** [comment]
Every single button/link on the https://reactflow.dev/ website suffers from this issue.
I just wanted to flag that this is a pretty serious accessibility issue and should probably be prioritised.
I'd be happy to look into it if anyone more familiar with the codebase can point me in the right direction.

**Comment by moklick** [comment]
An easy solution for this would be to treat buttons like inputs. There we do not handle key presses.

**Comment by moklick** [comment]
This is fixed in 12.5.0. We are not prevent the default behaviour if a button or link is focused.

**Comment by gkurt** [comment]
Is there a way to disable swallowing the Space key completely? This doesn't only happen with `button` or `input`, but any element can potentially handle Space key. I think XYFlow shouldn't swallow the event unless the focused element is inside the XYFlow root.

**Comment by bcakmakoglu** [comment]
@KurtGokhan Should be possible by assigning `panActivationKeyCode` to something other than `Space`

**Comment by Haeri** [comment]
Cool cool cool. Just took me an evening worth of debugging to figure out why my monaco editor was suddenly refusing to type space 😑

**Comment by neekfenwick** [comment]
I'd like to mention https://github.com/xyflow/xyflow/discussions/5690 where I posted a recent codesandbox (https://codesandbox.io/p/github/neekfenwick/TestReactApp/master) using xyflow 12.10.0 which still seems to demonstrate this problem, despite @moklick saying it was fixed in 12.5.0.  I found that adding `className="nokey"` to the Editor does fix the problem.

It's my first week or two using both Monaco Editor and XYFlow so I'm really not sure how to go about debugging this myself, just wanted to provide another example of the problem.  Thanks!

**Comment by omerchn** [comment]
Hey, this also happened to us and "nokey" fixed it. It seems this issue is not fixed yet. Can you re-open this issue? 🤔
